Index: source/smbd/mangle_hash2.c
===================================================================
RCS file: /cvsroot/samba/source/smbd/mangle_hash2.c,v
retrieving revision 1.16
retrieving revision 1.17
diff -u -w -r1.16 -r1.17
--- source/smbd/mangle_hash2.c	31 Aug 2002 16:17:36 -0000	1.16
+++ source/smbd/mangle_hash2.c	15 Oct 2002 05:40:24 -0000	1.17
@@ -147,14 +147,12 @@
 {
 	if (prefix_cache) return True;
 
-	prefix_cache = malloc(sizeof(char *) * MANGLE_CACHE_SIZE);
+	prefix_cache = calloc(MANGLE_CACHE_SIZE, sizeof(char *));
 	if (!prefix_cache) return False;
 
-	prefix_cache_hashes = malloc(sizeof(u32) * MANGLE_CACHE_SIZE);
+	prefix_cache_hashes = calloc(MANGLE_CACHE_SIZE, sizeof(u32));
 	if (!prefix_cache_hashes) return False;
 
-	memset(prefix_cache, 0, sizeof(char *) * MANGLE_CACHE_SIZE);
-	memset(prefix_cache_hashes, 0, sizeof(char *) * MANGLE_CACHE_SIZE);
 	return True;
 }
 
