diff -uNr samba-2.999+3.0cvs20020723.orig/source/Makefile.in samba-2.999+3.0cvs20020723/source/Makefile.in
--- samba-2.999+3.0cvs20020723.orig/source/Makefile.in	Thu Jul 25 10:15:35 2002
+++ samba-2.999+3.0cvs20020723/source/Makefile.in	Thu Jul 25 10:15:22 2002
@@ -45,6 +45,7 @@
 LIBDIR = @libdir@
 VARDIR = @localstatedir@
 MANDIR = @mandir@
+DATADIR = @datadir@
 
 # The permissions to give the executables
 INSTALLPERMS = 0755
@@ -68,6 +69,13 @@
 # the directory where lock files go
 LOCKDIR = @lockdir@
 
+# FHS directories; equal to LOCKDIR if not using --with-fhs
+CACHEDIR = @cachedir@
+STATEDIR = @statedir@
+
+# Where to look for (and install) codepage databases.
+CODEPAGEDIR = @codepagedir@
+
 # the directorty where pid files go
 PIDDIR = @piddir@
 # man pages language(s)
@@ -91,7 +99,8 @@
 PATH_FLAGS3 = $(PATH_FLAGS2) -DLMHOSTSFILE=\"$(LMHOSTSFILE)\" 
 PATH_FLAGS4 = $(PATH_FLAGS3) -DSWATDIR=\"$(SWATDIR)\"  -DLOCKDIR=\"$(LOCKDIR)\" -DPIDDIR=\"$(PIDDIR)\"
 PATH_FLAGS5 = $(PATH_FLAGS4) -DLIBDIR=\"$(LIBDIR)\" -DLOGFILEBASE=\"$(LOGFILEBASE)\"
-PATH_FLAGS = $(PATH_FLAGS5) $(PASSWD_FLAGS)
+PATH_FLAGS6 = $(PATH_FLAGS5) -DCODEPAGEDIR=\"$(CODEPAGEDIR)\" -DCACHEDIR=\"$(CACHEDIR)\" -DSTATEDIR=\"$(STATEDIR)\"
+PATH_FLAGS = $(PATH_FLAGS6) $(PASSWD_FLAGS)
 
 WINBIND_PROGS = @WINBIND_TARGETS@
 WINBIND_SPROGS = @WINBIND_STARGETS@
@@ -828,7 +837,7 @@
 	@$(SHELL) $(srcdir)/script/installscripts.sh $(INSTALLPERMS) $(BINDIR) $(SCRIPTS)
 
 installdat: installdirs
-	@$(SHELL) $(srcdir)/script/installdat.sh $(LIBDIR) $(srcdir)
+	@$(SHELL) $(srcdir)/script/installdat.sh $(CODEPAGEDIR) $(srcdir)
 
 installswat: installdirs
 	@$(SHELL) $(srcdir)/script/installswat.sh $(SWATDIR) $(srcdir)
diff -uNr samba-2.999+3.0cvs20020723.orig/source/acconfig.h samba-2.999+3.0cvs20020723/source/acconfig.h
--- samba-2.999+3.0cvs20020723.orig/source/acconfig.h	Mon Jul 22 22:39:55 2002
+++ samba-2.999+3.0cvs20020723/source/acconfig.h	Thu Jul 25 10:15:22 2002
@@ -188,6 +188,7 @@
 #undef STAT_ST_BLOCKSIZE
 #undef HAVE_DEVICE_MAJOR_FN
 #undef HAVE_DEVICE_MINOR_FN
+#undef FHS_COMPATIBLE
 #undef HAVE_PASSWD_PW_COMMENT
 #undef HAVE_PASSWD_PW_AGE
 /*
diff -uNr samba-2.999+3.0cvs20020723.orig/source/configure.in samba-2.999+3.0cvs20020723/source/configure.in
--- samba-2.999+3.0cvs20020723.orig/source/configure.in	Thu Jul 25 10:15:39 2002
+++ samba-2.999+3.0cvs20020723/source/configure.in	Thu Jul 25 10:15:22 2002
@@ -12,17 +12,24 @@
 AC_ARG_WITH(fhs, 
 [  --with-fhs              Use FHS-compliant paths (default=no)],
     configdir="${sysconfdir}/samba"
-    lockdir="\${VARDIR}/cache/samba"
+    lockdir="\${VARDIR}/run/samba"
     piddir="\$(VARDIR)/run/samba"
     logfilebase="\${VARDIR}/log/samba"
     privatedir="\${CONFIGDIR}/private"
-    swatdir="\${DATADIR}/samba/swat",
+    swatdir="\${DATADIR}/samba/swat"
+    codepagedir="\${DATADIR}/samba"
+    statedir="\${VARDIR}/lib/samba"
+    cachedir="\${VARDIR}/cache/samba"
+    AC_DEFINE(FHS_COMPATIBLE),
     configdir="\$(LIBDIR)"
     logfilebase="\$(VARDIR)"
     lockdir="\${VARDIR}/locks"
     piddir="\$(VARDIR)/locks"
     privatedir="\${prefix}/private"
-    swatdir="\${prefix}/swat")
+    codepagedir="\${LIBDIR}"
+    swatdir="\${prefix}/swat"
+    statedir="\${LOCKDIR}"
+    cachedir="\${LOCKDIR}")
 
 #################################################
 # set private directory location
@@ -128,6 +135,9 @@
 AC_SUBST(swatdir)
 AC_SUBST(bindir)
 AC_SUBST(sbindir)
+AC_SUBST(codepagedir)
+AC_SUBST(statedir)
+AC_SUBST(cachedir)
 
 dnl Unique-to-Samba variables we'll be playing with.
 AC_SUBST(SHELL)
diff -uNr samba-2.999+3.0cvs20020723.orig/source/dynconfig.c samba-2.999+3.0cvs20020723/source/dynconfig.c
--- samba-2.999+3.0cvs20020723.orig/source/dynconfig.c	Wed May  8 23:11:58 2002
+++ samba-2.999+3.0cvs20020723/source/dynconfig.c	Thu Jul 25 10:15:22 2002
@@ -52,11 +52,11 @@
 pstring dyn_LMHOSTSFILE = LMHOSTSFILE;
 
 /**
- * @brief Samba library directory.
+ * @brief Samba data directory.
  *
- * @sa lib_path() to get the path to a file inside the LIBDIR.
+ * @sa data_path() to get the path to a file inside the CODEPAGEDIR.
  **/
-pstring dyn_LIBDIR = LIBDIR;  
+pstring dyn_CODEPAGEDIR = CODEPAGEDIR;
 
 /**
  * @brief Directory holding lock files.
@@ -72,3 +72,25 @@
 const pstring dyn_PRIVATE_DIR = PRIVATE_DIR;
 
 
+/* In non-FHS mode, these should be configurable using 'lock dir =';
+   but in FHS mode, they are their own directory.  Implement as wrapper
+   functions so that everything can still be kept in dynconfig.c.
+ */
+
+char *dyn_STATEDIR(void)
+{
+#ifdef FHS_COMPATIBLE
+	return STATEDIR;
+#else
+	return lp_lockdir();
+#endif
+}
+
+char *dyn_CACHEDIR(void)
+{
+#ifdef FHS_COMPATIBLE
+	return CACHEDIR;
+#else
+	return lp_lockdir();
+#endif
+}
diff -uNr samba-2.999+3.0cvs20020723.orig/source/groupdb/mapping.c samba-2.999+3.0cvs20020723/source/groupdb/mapping.c
--- samba-2.999+3.0cvs20020723.orig/source/groupdb/mapping.c	Fri Jun  7 09:33:30 2002
+++ samba-2.999+3.0cvs20020723/source/groupdb/mapping.c	Thu Jul 25 10:15:22 2002
@@ -214,7 +214,7 @@
 	
 	if (tdb && local_pid == sys_getpid())
 		return True;
-	tdb = tdb_open_log(lock_path("group_mapping.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+	tdb = tdb_open_log(state_path("group_mapping.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!tdb) {
 		DEBUG(0,("Failed to open group mapping database\n"));
 		return False;
diff -uNr samba-2.999+3.0cvs20020723.orig/source/include/config.h.in samba-2.999+3.0cvs20020723/source/include/config.h.in
--- samba-2.999+3.0cvs20020723.orig/source/include/config.h.in	Mon Jul 22 22:40:40 2002
+++ samba-2.999+3.0cvs20020723/source/include/config.h.in	Thu Jul 25 10:15:22 2002
@@ -254,6 +254,7 @@
 #undef STAT_ST_BLOCKSIZE
 #undef HAVE_DEVICE_MAJOR_FN
 #undef HAVE_DEVICE_MINOR_FN
+#undef FHS_COMPATIBLE
 #undef HAVE_PASSWD_PW_COMMENT
 #undef HAVE_PASSWD_PW_AGE
 /*
diff -uNr samba-2.999+3.0cvs20020723.orig/source/include/dynconfig.h samba-2.999+3.0cvs20020723/source/include/dynconfig.h
--- samba-2.999+3.0cvs20020723.orig/source/include/dynconfig.h	Wed May  8 23:11:58 2002
+++ samba-2.999+3.0cvs20020723/source/include/dynconfig.h	Thu Jul 25 10:15:22 2002
@@ -29,9 +29,12 @@
 
 extern pstring dyn_CONFIGFILE;
 extern pstring dyn_LOGFILEBASE, dyn_LMHOSTSFILE;
-extern pstring dyn_LIBDIR;
+extern pstring dyn_CODEPAGEDIR;
 extern const pstring dyn_LOCKDIR; 
 extern const pstring dyn_PIDDIR;
 extern const pstring dyn_DRIVERFILE; 
 extern const pstring dyn_SMB_PASSWD_FILE;
 extern const pstring dyn_PRIVATE_DIR;
+
+char *dyn_STATEDIR();
+char *dyn_CACHEDIR();
diff -uNr samba-2.999+3.0cvs20020723.orig/source/intl/lang_tdb.c samba-2.999+3.0cvs20020723/source/intl/lang_tdb.c
--- samba-2.999+3.0cvs20020723.orig/source/intl/lang_tdb.c	Wed Jan 30 00:08:16 2002
+++ samba-2.999+3.0cvs20020723/source/intl/lang_tdb.c	Thu Jul 25 10:15:22 2002
@@ -121,7 +121,7 @@
 	/* if no lang then we don't translate */
 	if (!lang) return True;
 
-	asprintf(&msg_path, "%s.msg", lib_path((char *)lang));
+	asprintf(&msg_path, "%s.msg", data_path((char *)lang));
 	if (stat(msg_path, &st) != 0) {
 		/* the msg file isn't available */
 		free(msg_path);
@@ -129,7 +129,7 @@
 	}
 	
 
-	asprintf(&path, "%s%s.tdb", lock_path("lang_"), lang);
+	asprintf(&path, "%s%s.tdb", state_path("lang_"), lang);
 
 	tdb = tdb_open_log(path, 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0644);
 	if (!tdb) {
diff -uNr samba-2.999+3.0cvs20020723.orig/source/lib/account_pol.c samba-2.999+3.0cvs20020723/source/lib/account_pol.c
--- samba-2.999+3.0cvs20020723.orig/source/lib/account_pol.c	Wed Jan 30 00:08:16 2002
+++ samba-2.999+3.0cvs20020723/source/lib/account_pol.c	Thu Jul 25 10:15:22 2002
@@ -34,7 +34,7 @@
 
 	if (tdb && local_pid == sys_getpid())
 		return True;
-	tdb = tdb_open_log(lock_path("account_policy.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+	tdb = tdb_open_log(state_path("account_policy.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!tdb) {
 		DEBUG(0,("Failed to open account policy database\n"));
 		return False;
diff -uNr samba-2.999+3.0cvs20020723.orig/source/lib/util.c samba-2.999+3.0cvs20020723/source/lib/util.c
--- samba-2.999+3.0cvs20020723.orig/source/lib/util.c	Sat Jul 20 22:26:09 2002
+++ samba-2.999+3.0cvs20020723/source/lib/util.c	Thu Jul 25 10:15:22 2002
@@ -1904,16 +1904,56 @@
 
 
 /**
- * @brief Returns an absolute path to a file in the Samba lib directory.
+ * @brief Returns an absolute path to a file in the Samba data directory.
  *
- * @param name File to find, relative to LIBDIR.
+ * @param name File to find, relative to CODEPAGEDIR.
  *
  * @retval Pointer to a static #pstring containing the full path.
  **/
-char *lib_path(const char *name)
+char *data_path(const char *name)
 {
 	static pstring fname;
-	snprintf(fname, sizeof(fname), "%s/%s", dyn_LIBDIR, name);
+	snprintf(fname, sizeof(fname), "%s/%s", dyn_CODEPAGEDIR, name);
+	return fname;
+}
+
+/*****************************************************************
+a useful function for returning a path in the Samba state directory
+ *****************************************************************/
+char *state_path(char *name)
+{
+	static pstring fname;
+
+	pstrcpy(fname,dyn_STATEDIR());
+	trim_string(fname,"","/");
+
+	if (!directory_exist(fname,NULL)) {
+		mkdir(fname,0755);
+	}
+
+	pstrcat(fname,"/");
+	pstrcat(fname,name);
+
+	return fname;
+}
+
+/*****************************************************************
+a useful function for returning a path in the Samba cache directory
+ *****************************************************************/
+char *cache_path(char *name)
+{
+	static pstring fname;
+
+	pstrcpy(fname,dyn_CACHEDIR());
+	trim_string(fname,"","/");
+
+	if (!directory_exist(fname,NULL)) {
+		mkdir(fname,0755);
+	}
+
+	pstrcat(fname,"/");
+	pstrcat(fname,name);
+
 	return fname;
 }
 
diff -uNr samba-2.999+3.0cvs20020723.orig/source/lib/util_unistr.c samba-2.999+3.0cvs20020723/source/lib/util_unistr.c
--- samba-2.999+3.0cvs20020723.orig/source/lib/util_unistr.c	Sun Apr 14 04:44:14 2002
+++ samba-2.999+3.0cvs20020723/source/lib/util_unistr.c	Thu Jul 25 10:15:22 2002
@@ -43,8 +43,8 @@
 	if (initialised) return;
 	initialised = 1;
 
-	upcase_table = map_file(lib_path("upcase.dat"), 0x20000);
-	lowcase_table = map_file(lib_path("lowcase.dat"), 0x20000);
+	upcase_table = map_file(data_path("upcase.dat"), 0x20000);
+	lowcase_table = map_file(data_path("lowcase.dat"), 0x20000);
 
 	/* we would like Samba to limp along even if these tables are
 	   not available */
@@ -108,7 +108,7 @@
 	if (initialised && mapped_file) return;
 	initialised = 1;
 
-	valid_table = map_file(lib_path("valid.dat"), 0x10000);
+	valid_table = map_file(data_path("valid.dat"), 0x10000);
 	if (valid_table) {
 		mapped_file = 1;
 		return;
diff -uNr samba-2.999+3.0cvs20020723.orig/source/libsmb/netlogon_unigrp.c samba-2.999+3.0cvs20020723/source/libsmb/netlogon_unigrp.c
--- samba-2.999+3.0cvs20020723.orig/source/libsmb/netlogon_unigrp.c	Fri Mar 15 03:19:51 2002
+++ samba-2.999+3.0cvs20020723/source/libsmb/netlogon_unigrp.c	Thu Jul 25 10:15:22 2002
@@ -40,7 +40,7 @@
 BOOL uni_group_cache_init(void)
 {
 	if (!netlogon_unigrp_tdb) {
-		netlogon_unigrp_tdb = tdb_open_log(lock_path("netlogon_unigrp.tdb"), 0,
+		netlogon_unigrp_tdb = tdb_open_log(state_path("netlogon_unigrp.tdb"), 0,
 						   TDB_DEFAULT, O_RDWR | O_CREAT, 0644);
 	}
 
@@ -110,7 +110,7 @@
 		return NULL;
 	}
 	if (!netlogon_unigrp_tdb) {
-		netlogon_unigrp_tdb = tdb_open_log(lock_path("netlogon_unigrp.tdb"), 0,
+		netlogon_unigrp_tdb = tdb_open_log(state_path("netlogon_unigrp.tdb"), 0,
                 				    TDB_DEFAULT, O_RDWR, 0644);
 	}
 	if (!netlogon_unigrp_tdb) {
diff -uNr samba-2.999+3.0cvs20020723.orig/source/nmbd/nmbd_serverlistdb.c samba-2.999+3.0cvs20020723/source/nmbd/nmbd_serverlistdb.c
--- samba-2.999+3.0cvs20020723.orig/source/nmbd/nmbd_serverlistdb.c	Wed Jan 30 00:08:22 2002
+++ samba-2.999+3.0cvs20020723/source/nmbd/nmbd_serverlistdb.c	Thu Jul 25 10:15:22 2002
@@ -347,7 +347,7 @@
 
   updatecount++;
     
-  pstrcpy(fname,lp_lockdir());
+  pstrcpy(fname,dyn_CACHEDIR());
   trim_string(fname,NULL,"/");
   pstrcat(fname,"/");
   pstrcat(fname,SERVER_LIST);
diff -uNr samba-2.999+3.0cvs20020723.orig/source/nmbd/nmbd_winsserver.c samba-2.999+3.0cvs20020723/source/nmbd/nmbd_winsserver.c
--- samba-2.999+3.0cvs20020723.orig/source/nmbd/nmbd_winsserver.c	Mon Jul  1 02:19:12 2002
+++ samba-2.999+3.0cvs20020723/source/nmbd/nmbd_winsserver.c	Thu Jul 25 10:16:28 2002
@@ -234,7 +234,7 @@
 
 	add_samba_names_to_subnet(wins_server_subnet);
 
-	tdb = tdb_open_log(lock_path(WINS_LIST), 0, TDB_DEFAULT, O_RDONLY, 0600);
+	tdb = tdb_open_log(state_path(WINS_LIST), 0, TDB_DEFAULT, O_RDONLY, 0600);
 	if (!tdb) {
 		DEBUG(2,("initialise_wins: Can't open wins database file %s. Error was %s\n", WINS_LIST, strerror(errno) ));
 		return True;
@@ -1762,7 +1762,7 @@
 		}
 	}
 
-	slprintf(fname,sizeof(fname)-1,"%s/%s", lp_lockdir(), WINS_LIST);
+	slprintf(fname,sizeof(fname)-1,"%s/%s", dyn_STATEDIR(), WINS_LIST);
 	all_string_sub(fname,"//", "/", 0);
 	slprintf(fnamenew,sizeof(fnamenew)-1,"%s.%u", fname, (unsigned int)sys_getpid());
 
diff -uNr samba-2.999+3.0cvs20020723.orig/source/nsswitch/winbindd_ads.c samba-2.999+3.0cvs20020723/source/nsswitch/winbindd_ads.c
--- samba-2.999+3.0cvs20020723.orig/source/nsswitch/winbindd_ads.c	Thu Jul 11 00:28:08 2002
+++ samba-2.999+3.0cvs20020723/source/nsswitch/winbindd_ads.c	Thu Jul 25 10:15:22 2002
@@ -116,7 +116,7 @@
 	}
 
 	/* we don't want this to affect the users ccache */
-	ccache = lock_path("winbindd_ccache");
+	ccache = cache_path("winbindd_ccache");
 	SETENV("KRB5CCNAME", ccache, 1);
 	unlink(ccache);
 
diff -uNr samba-2.999+3.0cvs20020723.orig/source/nsswitch/winbindd_cache.c samba-2.999+3.0cvs20020723/source/nsswitch/winbindd_cache.c
--- samba-2.999+3.0cvs20020723.orig/source/nsswitch/winbindd_cache.c	Tue Jun 18 04:20:08 2002
+++ samba-2.999+3.0cvs20020723/source/nsswitch/winbindd_cache.c	Thu Jul 25 10:15:22 2002
@@ -53,7 +53,7 @@
 	}
 	if (opt_nocache) return;
 
-	wcache->tdb = tdb_open_log(lock_path("winbindd_cache.tdb"), 5000, 
+	wcache->tdb = tdb_open_log(cache_path("winbindd_cache.tdb"), 5000, 
 				   TDB_CLEAR_IF_FIRST, O_RDWR|O_CREAT, 0600);
 
 	if (!wcache->tdb) {
diff -uNr samba-2.999+3.0cvs20020723.orig/source/nsswitch/winbindd_idmap.c samba-2.999+3.0cvs20020723/source/nsswitch/winbindd_idmap.c
--- samba-2.999+3.0cvs20020723.orig/source/nsswitch/winbindd_idmap.c	Tue Jun 18 04:20:09 2002
+++ samba-2.999+3.0cvs20020723/source/nsswitch/winbindd_idmap.c	Thu Jul 25 10:15:23 2002
@@ -430,14 +430,14 @@
 {
 	/* Open tdb cache */
 
-	if (!(idmap_tdb = tdb_open_log(lock_path("winbindd_idmap.tdb"), 0,
+	if (!(idmap_tdb = tdb_open_log(state_path("winbindd_idmap.tdb"), 0,
 				TDB_DEFAULT, O_RDWR | O_CREAT, 0600))) {
 		DEBUG(0, ("winbindd_idmap_init: Unable to open idmap database\n"));
 		return False;
 	}
 
 	/* possibly convert from an earlier version */
-	if (!idmap_convert(lock_path("winbindd_idmap.tdb"))) {
+	if (!idmap_convert(state_path("winbindd_idmap.tdb"))) {
 		DEBUG(0, ("winbindd_idmap_init: Unable to open idmap database\n"));
 		return False;
 	}
diff -uNr samba-2.999+3.0cvs20020723.orig/source/param/loadparm.c samba-2.999+3.0cvs20020723/source/param/loadparm.c
--- samba-2.999+3.0cvs20020723.orig/source/param/loadparm.c	Sat Jul 20 18:55:17 2002
+++ samba-2.999+3.0cvs20020723/source/param/loadparm.c	Thu Jul 25 10:15:23 2002
@@ -95,6 +95,9 @@
 	char *szAddPrinterCommand;
 	char *szDeletePrinterCommand;
 	char *szOs2DriverMap;
+#ifdef FHS_COMPATIBLE
+	char *szLockDirStub;
+#endif
 	char *szLockDir;
 	char *szPidDir;
 	char *szRootdir;
@@ -966,8 +969,13 @@
 	{"config file", P_STRING, P_GLOBAL, &Globals.szConfigFile, NULL, NULL, FLAG_HIDE},
 	{"preload", P_STRING, P_GLOBAL, &Globals.szAutoServices, NULL, NULL, 0},
 	{"auto services", P_STRING, P_GLOBAL, &Globals.szAutoServices, NULL, NULL, 0},
+#ifdef FHS_COMPATIBLE
+	{"lock dir", P_STRING, P_GLOBAL, &Globals.szLockDirStub, NULL, NULL, 0},
+	{"lock directory", P_STRING, P_GLOBAL, &Globals.szLockDirStub, NULL, NULL, 0},
+#else
 	{"lock dir", P_STRING, P_GLOBAL, &Globals.szLockDir, NULL, NULL, 0}, 
 	{"lock directory", P_STRING, P_GLOBAL, &Globals.szLockDir, NULL, NULL, 0},
+#endif
 	{"pid directory", P_STRING, P_GLOBAL, &Globals.szPidDir, NULL, NULL, 0}, 
 #ifdef WITH_UTMP
 	{"utmp directory", P_STRING, P_GLOBAL, &Globals.szUtmpDir, NULL, NULL, 0},
diff -uNr samba-2.999+3.0cvs20020723.orig/source/passdb/secrets.c samba-2.999+3.0cvs20020723/source/passdb/secrets.c
--- samba-2.999+3.0cvs20020723.orig/source/passdb/secrets.c	Sat Jul 20 08:02:46 2002
+++ samba-2.999+3.0cvs20020723/source/passdb/secrets.c	Thu Jul 25 10:15:23 2002
@@ -37,8 +37,7 @@
 	if (tdb)
 		return True;
 
-	pstrcpy(fname, lp_private_dir());
-	pstrcat(fname,"/secrets.tdb");
+	pstrcpy(fname,state_path("secrets.tdb"));
 
 	tdb = tdb_open_log(fname, 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 
diff -uNr samba-2.999+3.0cvs20020723.orig/source/printing/nt_printing.c samba-2.999+3.0cvs20020723/source/printing/nt_printing.c
--- samba-2.999+3.0cvs20020723.orig/source/printing/nt_printing.c	Sat Jul 20 08:23:56 2002
+++ samba-2.999+3.0cvs20020723/source/printing/nt_printing.c	Thu Jul 25 10:15:23 2002
@@ -263,24 +263,24 @@
 	if (tdb_drivers && tdb_printers && tdb_forms && local_pid == sys_getpid())
 		return True;
  
-	tdb_drivers = tdb_open_log(lock_path("ntdrivers.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+	tdb_drivers = tdb_open_log(state_path("ntdrivers.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!tdb_drivers) {
 		DEBUG(0,("nt_printing_init: Failed to open nt drivers database %s (%s)\n",
-			lock_path("ntdrivers.tdb"), strerror(errno) ));
+			state_path("ntdrivers.tdb"), strerror(errno) ));
 		return False;
 	}
  
-	tdb_printers = tdb_open_log(lock_path("ntprinters.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+	tdb_printers = tdb_open_log(state_path("ntprinters.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!tdb_printers) {
 		DEBUG(0,("nt_printing_init: Failed to open nt printers database %s (%s)\n",
-			lock_path("ntprinters.tdb"), strerror(errno) ));
+			state_path("ntprinters.tdb"), strerror(errno) ));
 		return False;
 	}
  
-	tdb_forms = tdb_open_log(lock_path("ntforms.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+	tdb_forms = tdb_open_log(state_path("ntforms.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!tdb_forms) {
 		DEBUG(0,("nt_printing_init: Failed to open nt forms database %s (%s)\n",
-			lock_path("ntforms.tdb"), strerror(errno) ));
+			state_path("ntforms.tdb"), strerror(errno) ));
 		return False;
 	}
  
diff -uNr samba-2.999+3.0cvs20020723.orig/source/printing/printing.c samba-2.999+3.0cvs20020723/source/printing/printing.c
--- samba-2.999+3.0cvs20020723.orig/source/printing/printing.c	Mon Jul 22 16:40:39 2002
+++ samba-2.999+3.0cvs20020723/source/printing/printing.c	Thu Jul 25 10:15:23 2002
@@ -185,7 +185,7 @@
 		DLIST_ADD(print_db_head, p);
 	}
 
-	pstrcpy(printdb_path, lock_path("printing/"));
+	pstrcpy(printdb_path, cache_path("printing/"));
 	pstrcat(printdb_path, printername);
 	pstrcat(printdb_path, ".tdb");
 
@@ -219,8 +219,8 @@
 	if (local_pid == sys_getpid())
 		return True;
 
-	unlink(lock_path("printing.tdb"));
-	pstrcpy(printing_path,lock_path("printing"));
+	unlink(cache_path("printing.tdb"));
+	pstrcpy(printing_path,cache_path("printing"));
 	mkdir(printing_path,0755);
 
 	local_pid = sys_getpid();
diff -uNr samba-2.999+3.0cvs20020723.orig/source/registry/reg_db.c samba-2.999+3.0cvs20020723/source/registry/reg_db.c
--- samba-2.999+3.0cvs20020723.orig/source/registry/reg_db.c	Fri Jul 19 23:27:30 2002
+++ samba-2.999+3.0cvs20020723/source/registry/reg_db.c	Thu Jul 25 10:15:23 2002
@@ -125,13 +125,13 @@
 	 * if we need to init the data in the registry
 	 */
 	
-	tdb_reg = tdb_open_log(lock_path("registry.tdb"), 0, TDB_DEFAULT, O_RDWR, 0600);
+	tdb_reg = tdb_open_log(state_path("registry.tdb"), 0, TDB_DEFAULT, O_RDWR, 0600);
 	if ( !tdb_reg ) 
 	{
-		tdb_reg = tdb_open_log(lock_path("registry.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+		tdb_reg = tdb_open_log(state_path("registry.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 		if ( !tdb_reg ) {
 			DEBUG(0,("init_registry: Failed to open registry %s (%s)\n",
-				lock_path("registry.tdb"), strerror(errno) ));
+				state_path("registry.tdb"), strerror(errno) ));
 			return False;
 		}
 		
diff -uNr samba-2.999+3.0cvs20020723.orig/source/rpc_server/srv_srvsvc_nt.c samba-2.999+3.0cvs20020723/source/rpc_server/srv_srvsvc_nt.c
--- samba-2.999+3.0cvs20020723.orig/source/rpc_server/srv_srvsvc_nt.c	Mon Jul 15 05:54:34 2002
+++ samba-2.999+3.0cvs20020723/source/rpc_server/srv_srvsvc_nt.c	Thu Jul 25 10:15:23 2002
@@ -127,10 +127,10 @@
  
 	if (share_tdb && local_pid == sys_getpid())
 		return True;
-	share_tdb = tdb_open_log(lock_path("share_info.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+	share_tdb = tdb_open_log(state_path("share_info.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!share_tdb) {
 		DEBUG(0,("Failed to open share info database %s (%s)\n",
-			lock_path("share_info.tdb"), strerror(errno) ));
+			state_path("share_info.tdb"), strerror(errno) ));
 		return False;
 	}
  
diff -uNr samba-2.999+3.0cvs20020723.orig/source/smbd/lanman.c samba-2.999+3.0cvs20020723/source/smbd/lanman.c
--- samba-2.999+3.0cvs20020723.orig/source/smbd/lanman.c	Sat Jul 20 22:34:02 2002
+++ samba-2.999+3.0cvs20020723/source/smbd/lanman.c	Thu Jul 25 10:15:23 2002
@@ -1095,9 +1095,9 @@
   BOOL local_list_only;
   int i;
 
-  lines = file_lines_load(lock_path(SERVER_LIST), NULL);
+  lines = file_lines_load(cache_path(SERVER_LIST), NULL);
   if (!lines) {
-    DEBUG(4,("Can't open %s - %s\n",lock_path(SERVER_LIST),strerror(errno)));
+    DEBUG(4,("Can't open %s - %s\n",cache_path(SERVER_LIST),strerror(errno)));
     return(0);
   }
 
diff -uNr samba-2.999+3.0cvs20020723.orig/source/wrepld/process.c samba-2.999+3.0cvs20020723/source/wrepld/process.c
--- samba-2.999+3.0cvs20020723.orig/source/wrepld/process.c	Fri Jun 14 11:02:59 2002
+++ samba-2.999+3.0cvs20020723/source/wrepld/process.c	Thu Jul 25 10:15:23 2002
@@ -197,7 +197,7 @@
 {
 	TDB_CONTEXT *tdb;
 
-	tdb = tdb_open_log(lock_path(WINS_LIST), 0, TDB_DEFAULT, O_RDONLY, 0600);
+	tdb = tdb_open_log(state_path(WINS_LIST), 0, TDB_DEFAULT, O_RDONLY, 0600);
 	if (!tdb) {
 		DEBUG(2,("get_our_last_id: Can't open wins database file %s. Error was %s\n", WINS_LIST, strerror(errno) ));
 		return;
@@ -489,7 +489,7 @@
 		}
 
 
-	tdb = tdb_open_log(lock_path(WINS_LIST), 0, TDB_DEFAULT, O_RDONLY, 0600);
+	tdb = tdb_open_log(state_path(WINS_LIST), 0, TDB_DEFAULT, O_RDONLY, 0600);
 	if (!tdb) {
 		DEBUG(2,("send_entry_request: Can't open wins database file %s. Error was %s\n", WINS_LIST, strerror(errno) ));
 		return;
