--- samba-3.0.7/source/passdb/pdb_ldap.c.orig 2004-09-29
22:57:47.000000000 +0100
+++ samba-3.0.7/source/passdb/pdb_ldap.c 2004-09-29
22:58:18.000000000 +0100
@@ -1253,7 +1253,8 @@
  int rc;
  
  attr_list = get_userattr_list( ldap_state->schema_ver );
- append_attr(&attr_list, get_userattr_key2string(ldap_state->schema_ver,LDAP_ATTR_MOD_TIMESTAMP));
+ if( ldap_state->schema_ver == SCHEMAVER_SAMBASAMACCOUNT )
+      append_attr(&attr_list, get_userattr_key2string(ldap_state->schema_ver,LDAP_ATTR_MOD_TIMESTAMP));
  rc = ldapsam_search_suffix_by_name(ldap_state, sname, &result, attr_list);
  free_attr_list( attr_list );
 
