diff -uNr samba-2.999+3.0cvs20020805.orig/source/configure.in samba-2.999+3.0cvs20020805/source/configure.in
--- samba-2.999+3.0cvs20020805.orig/source/configure.in	2002-08-20 15:34:29.000000000 -0500
+++ samba-2.999+3.0cvs20020805/source/configure.in	2002-08-20 15:34:44.000000000 -0500
@@ -157,6 +157,7 @@
 AC_SUBST(SHLIBEXT)
 AC_SUBST(LIBSMBCLIENT_SHARED)
 AC_SUBST(LIBSMBCLIENT)
+AC_SUBST(PRINTLIBS)
 
 # compile with optimization and without debugging by default
 CFLAGS="-O ${CFLAGS}"
@@ -528,7 +529,7 @@
                         AC_DEFINE(HAVE_CUPS)
 		CFLAGS="$CFLAGS `$CUPS_CONFIG --cflags`"
 		LDFLAGS="$LDFLAGS `$CUPS_CONFIG --ldflags`"
-		LIBS="$LIBS `$CUPS_CONFIG --libs`"
+		PRINTLIBS="$PRINTLIBS `$CUPS_CONFIG --libs`"
         fi
 fi
 
diff -uNr samba-2.999+3.0cvs20020805.orig/source/Makefile.in samba-2.999+3.0cvs20020805/source/Makefile.in
--- samba-2.999+3.0cvs20020805.orig/source/Makefile.in	2002-08-20 15:34:29.000000000 -0500
+++ samba-2.999+3.0cvs20020805/source/Makefile.in	2002-08-20 15:34:44.000000000 -0500
@@ -21,6 +21,7 @@
 
 TERMLDFLAGS=@TERMLDFLAGS@
 TERMLIBS=@TERMLIBS@
+PRINTLIBS=@PRINTLIBS@
 
 LINK=$(CC) $(FLAGS) $(LDFLAGS)
 
@@ -617,7 +618,7 @@
 
 bin/smbd: $(SMBD_OBJ) bin/.dummy
 	@echo Linking $@
-	@$(CC) $(FLAGS) -o $@ $(SMBD_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) 
+	@$(CC) $(FLAGS) -o $@ $(SMBD_OBJ) $(LDFLAGS) $(DYNEXP) $(PRINTLIBS) $(LIBS) 
 
 bin/nmbd: $(NMBD_OBJ) bin/.dummy
 	@echo Linking $@
@@ -629,7 +630,7 @@
 
 bin/swat: $(SWAT_OBJ) bin/.dummy
 	@echo Linking $@
-	@$(CC) $(FLAGS) -o $@ $(SWAT_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) 
+	@$(CC) $(FLAGS) -o $@ $(SWAT_OBJ) $(LDFLAGS) $(DYNEXP) $(PRINTLIBS) $(LIBS) 
 
 bin/rpcclient: $(RPCCLIENT_OBJ) @BUILD_POPT@ bin/.dummy
 	@echo Linking $@
@@ -669,7 +670,7 @@
 
 bin/testprns: $(TESTPRNS_OBJ) bin/.dummy
 	@echo Linking $@
-	@$(CC) $(FLAGS) -o $@ $(TESTPRNS_OBJ) $(LDFLAGS) $(LIBS)
+	@$(CC) $(FLAGS) -o $@ $(TESTPRNS_OBJ) $(LDFLAGS) $(PRINTLIBS) $(LIBS)
 
 bin/smbstatus: $(STATUS_OBJ) @BUILD_POPT@ bin/.dummy
 	@echo Linking $@
