diff -uNr samba-3.0alpha21.orig/source/Makefile.in samba-3.0alpha21/source/Makefile.in
--- samba-3.0alpha21.orig/source/Makefile.in	2003-02-16 13:40:12.000000000 -0600
+++ samba-3.0alpha21/source/Makefile.in	2003-02-16 13:40:25.000000000 -0600
@@ -26,6 +26,7 @@
 TERMLIBS=@TERMLIBS@
 PRINTLIBS=@PRINTLIBS@
 AUTHLIBS=@AUTHLIBS@
+ACLLIBS=@ACLLIBS@
 
 LINK=$(CC) $(FLAGS) $(LDFLAGS)
 
@@ -699,7 +700,7 @@
 bin/smbd@EXEEXT@: $(SMBD_OBJ) @BUILD_POPT@ bin/.dummy
 	@echo Linking $@
 	@$(CC) $(FLAGS) -o $@ $(SMBD_OBJ) $(LDFLAGS) $(DYNEXP) $(PRINTLIBS) \
-	  $(AUTHLIBS) $(LIBS) @BUILD_POPT@
+	  $(AUTHLIBS) $(ACLLIBS) $(LIBS) @BUILD_POPT@
 
 bin/nmbd@EXEEXT@: $(NMBD_OBJ) @BUILD_POPT@ bin/.dummy
 	@echo Linking $@
diff -uNr samba-3.0alpha21.orig/source/configure.in samba-3.0alpha21/source/configure.in
--- samba-3.0alpha21.orig/source/configure.in	2003-02-16 13:40:12.000000000 -0600
+++ samba-3.0alpha21/source/configure.in	2003-02-16 13:44:08.000000000 -0600
@@ -156,6 +156,7 @@
 AC_SUBST(LIBSMBCLIENT)
 AC_SUBST(PRINTLIBS)
 AC_SUBST(AUTHLIBS)
+AC_SUBST(ACLLIBS)
 AC_SUBST(SHLIB_PROGS)
 AC_SUBST(SMBWRAPPER)
 AC_SUBST(EXTRA_BIN_PROGS)
@@ -2938,23 +2939,29 @@
 	*osf*)
 		AC_MSG_RESULT(Using Tru64 ACLs)
 		AC_DEFINE(HAVE_TRU64_ACLS,1,[Whether Tru64 ACLs are available])
-		LIBS="$LIBS -lpacl"
+		ACLLIBS="$ACLLIBS -lpacl"
 		;;
         *)
-		AC_CHECK_LIB(acl,acl_get_file)
+		AC_CHECK_LIB(acl,acl_get_file,[ACLLIBS="$ACLLIBS -lacl"])
 		AC_CACHE_CHECK([for ACL support],samba_cv_HAVE_POSIX_ACLS,[
+		acl_LIBS=$LIBS
+		LIBS="$LIBS -lacl"
 		AC_TRY_LINK([#include <sys/types.h>
 #include <sys/acl.h>],
 [ acl_t acl; int entry_id; acl_entry_t *entry_p; return acl_get_entry( acl, entry_id, entry_p);],
-samba_cv_HAVE_POSIX_ACLS=yes,samba_cv_HAVE_POSIX_ACLS=no)])
+samba_cv_HAVE_POSIX_ACLS=yes,samba_cv_HAVE_POSIX_ACLS=no)
+		LIBS=$acl_LIBS])
 			if test x"$samba_cv_HAVE_POSIX_ACLS" = x"yes"; then
 				AC_MSG_RESULT(Using posix ACLs)
 				AC_DEFINE(HAVE_POSIX_ACLS,1,[Whether POSIX ACLs are available])
 				AC_CACHE_CHECK([for acl_get_perm_np],samba_cv_HAVE_ACL_GET_PERM_NP,[
+				acl_LIBS=$LIBS
+				LIBS="$LIBS -lacl"
 				AC_TRY_LINK([#include <sys/types.h>
 #include <sys/acl.h>],
 [ acl_permset_t permset_d; acl_perm_t perm; return acl_get_perm_np( permset_d, perm);],
-samba_cv_HAVE_ACL_GET_PERM_NP=yes,samba_cv_HAVE_ACL_GET_PERM_NP=no)])
+samba_cv_HAVE_ACL_GET_PERM_NP=yes,samba_cv_HAVE_ACL_GET_PERM_NP=no)
+				LIBS=$acl_LIBS])
 				if test x"$samba_cv_HAVE_ACL_GET_PERM_NP" = x"yes"; then
 					AC_DEFINE(HAVE_ACL_GET_PERM_NP,1,[Whether acl_get_perm_np() is available])
 				fi
