diff -ur samba-2.99.cvs.20020713.orig/source/include/libsmbclient.h samba-2.99.cvs.20020713/source/include/libsmbclient.h
--- samba-2.99.cvs.20020713.orig/source/include/libsmbclient.h	Sat Jul 13 02:18:41 2002
+++ samba-2.99.cvs.20020713/source/include/libsmbclient.h	Sat Jul 13 14:16:38 2002
@@ -51,7 +51,7 @@
 *   \ingroup libsmbclient
 *   Functions used to access printing functionality
 */
-/** \defgroup attribute Miscellaneous Functions
+/** \defgroup misc Miscellaneous Functions
 *   \ingroup libsmbclient
 *   Functions that don't fit in to other categories
 */
