diff -uNr samba-2.999+3.0.alpha21.orig/source/param/loadparm.c samba-2.999+3.0.alpha21/source/param/loadparm.c
--- samba-2.999+3.0.alpha21.orig/source/param/loadparm.c	2002-12-20 09:19:50.000000000 -0600
+++ samba-2.999+3.0.alpha21/source/param/loadparm.c	2002-12-20 09:31:09.000000000 -0600
@@ -1364,10 +1364,8 @@
 #ifdef WITH_LDAP_SAMCONFIG
 	string_set(&Globals.szLdapServer, "localhost");
 	Globals.ldap_port = 636;
-	Globals.szPassdbBackend = str_list_make("ldapsam unixsam", NULL);
-#else
-	Globals.szPassdbBackend = str_list_make("smbpasswd unixsam", NULL);
 #endif /* WITH_LDAP_SAMCONFIG */
+	Globals.szPassdbBackend = str_list_make("smbpasswd unixsam", NULL);
 
 	string_set(&Globals.szLdapSuffix, "");
 	string_set(&Globals.szLdapMachineSuffix, "");
