diff -uNr samba-3.0alpha21.orig/source/utils/pdbedit.c samba-3.0alpha21/source/utils/pdbedit.c
--- samba-3.0alpha21.orig/source/utils/pdbedit.c	2002-11-26 20:54:22.000000000 -0600
+++ samba-3.0alpha21/source/utils/pdbedit.c	2002-12-22 13:38:25.000000000 -0600
@@ -495,7 +495,7 @@
 	poptGetArg(pc); /* Drop argv[0], the program name */
 
 	if (user_name == NULL) {
-		user_name = strdup(poptGetArg(pc));
+		user_name = poptGetArg(pc);
 	}
 
 	if (!lp_load(dyn_CONFIGFILE,True,False,False)) {
@@ -503,6 +503,9 @@
 		exit(1);
 	}
 
+	if(lp_modules())
+		smb_load_modules(lp_modules());
+	
 	if (!init_names())
 		exit(1);
 
