diff -uNr samba-3.0alpha23.orig/source/printing/printing.c samba-3.0alpha23/source/printing/printing.c
--- samba-3.0alpha23.orig/source/printing/printing.c	2003-04-09 09:37:26.000000000 -0500
+++ samba-3.0alpha23/source/printing/printing.c	2003-04-09 09:39:21.000000000 -0500
@@ -835,7 +835,7 @@
 
 	/* Work out the size. */
 	data.dsize = 0;
-	data.dsize += tdb_pack(NULL, 0, NULL, "d", qcount);
+	data.dsize += tdb_pack(NULL, 0, "d", qcount);
 
 	for (i = 0; i < pts->qcount; i++) {
 		data.dsize += tdb_pack(NULL, 0, "ddddddff",
diff -uNr samba-3.0alpha23.orig/source/rpc_server/srv_spoolss_nt.c samba-3.0alpha23/source/rpc_server/srv_spoolss_nt.c
--- samba-3.0alpha23.orig/source/rpc_server/srv_spoolss_nt.c	2003-03-30 20:28:31.000000000 -0600
+++ samba-3.0alpha23/source/rpc_server/srv_spoolss_nt.c	2003-04-09 09:38:53.000000000 -0500
@@ -8324,7 +8324,6 @@
 	info_1=(JOB_INFO_1 *)malloc(sizeof(JOB_INFO_1));
 
 	if (info_1 == NULL) {
-		SAFE_FREE(queue);
 		return WERR_NOMEM;
 	}
 		
@@ -8334,7 +8333,6 @@
 	}
 	
 	if (found==False) {
-		SAFE_FREE(queue);
 		SAFE_FREE(info_1);
 		/* NT treats not found as bad param... yet another bad choice */
 		return WERR_INVALID_PARAM;
