diff -uNr samba-3.0alpha23.orig/source/Makefile.in samba-3.0alpha23/source/Makefile.in
--- samba-3.0alpha23.orig/source/Makefile.in	2003-04-14 23:15:09.000000000 -0500
+++ samba-3.0alpha23/source/Makefile.in	2003-04-14 23:30:23.000000000 -0500
@@ -1044,20 +1044,21 @@
 PYTHON_OBJS = $(LIB_OBJ) $(LIBSMB_OBJ) $(RPC_PARSE_OBJ) $(UBIQX_OBJ) \
 	$(PARAM_OBJ) $(LIBMSRPC_OBJ) $(PASSDB_OBJ) $(GROUPDB_OBJ) \
 	$(SECRETS_OBJ) $(KRBCLIENT_OBJ)
+PYTHON_PICOBJS = $(PYTHON_OBJS:.o=.po)
 
-python_ext: $(PYTHON_OBJS)
+python_ext: $(PYTHON_PICOBJS)
 	@if test -z "$(PYTHON)"; then \
 		echo Use the option --with-python to configure python; \
 		exit 1; fi
-	PYTHON_OBJS="$(PYTHON_OBJS)" PYTHON_CFLAGS="$(CFLAGS) $(CPPFLAGS) $(FLAGS)" \
+	PYTHON_OBJS="$(PYTHON_PICOBJS)" PYTHON_CFLAGS="$(CFLAGS) $(CPPFLAGS) $(FLAGS)" \
 	LIBS="$(LIBS)" \
 		$(PYTHON) python/setup.py build
 
-python_install: $(PYTHON_OBJS)
+python_install: $(PYTHON_PICOBJS)
 	@if test -z "$(PYTHON)"; then \
 		echo Use the option --with-python to configure python; \
 		exit 1; fi
-	PYTHON_OBJS="$(PYTHON_OBJS)" PYTHON_CFLAGS="$(CFLAGS) $(CPPFLAGS)" \
+	PYTHON_OBJS="$(PYTHON_PICOBJS)" PYTHON_CFLAGS="$(CFLAGS) $(CPPFLAGS)" \
 	LIBS="$(LIBS)" \
 		$(PYTHON) python/setup.py install
 
