diff -uNr samba-2.999+3.0.alpha20.orig/source/smbd/lanman.c samba-2.999+3.0.alpha20/source/smbd/lanman.c
--- samba-2.999+3.0.alpha20.orig/source/smbd/lanman.c	2002-10-04 14:56:51.000000000 -0500
+++ samba-2.999+3.0.alpha20/source/smbd/lanman.c	2002-10-04 23:06:54.000000000 -0500
@@ -72,7 +72,7 @@
 	StrnCpy(buf,src,sizeof(buf)/2);
 	pstring_sub(buf,"%S",lp_servicename(snum));
 	standard_sub_conn(conn,buf,sizeof(buf));
-	l = push_ascii(*dst,buf,*n-1, STR_TERMINATE);
+	l = push_ascii(*dst,buf,*n, STR_TERMINATE);
 	(*dst) += l;
 	(*n) -= l;
 	return l;
