diff -ur samba-2.99.cvs.20020713.orig/docs/docbook/manpages/smb.conf.5.sgml samba-2.99.cvs.20020713/docs/docbook/manpages/smb.conf.5.sgml
--- samba-2.99.cvs.20020713.orig/docs/docbook/manpages/smb.conf.5.sgml	Tue Jul  2 09:17:42 2002
+++ samba-2.99.cvs.20020713/docs/docbook/manpages/smb.conf.5.sgml	Sat Jul 13 14:37:24 2002
@@ -4532,7 +4532,7 @@
 
 		<para>Here's a way of sending the messages as mail to root:</para>
 
-		<para><command>message command = /bin/mail -s 'message from %f on 
+		<para><command>message command = /usr/bin/mail -s 'message from %f on 
 		%m' root &lt; %s; rm %s</command></para>
 
 		<para>If you don't have a message command then the message 
