diff -ur samba-2.99.cvs.20020713.orig/source/smbadduser samba-2.99.cvs.20020713/source/smbadduser
--- samba-2.99.cvs.20020713.orig/source/smbadduser	Mon Oct 13 07:13:11 1997
+++ samba-2.99.cvs.20020713/source/smbadduser	Sat Jul 13 15:02:34 2002
@@ -2,11 +2,14 @@
 #
 # smbadduser - Written by Mike Zakharoff
 #
+# Customized for Debian by Eloy A. Paris <peloy@debian.org>
+#
 unalias *
-set path = ($path /usr/local/samba/bin)
+# No need to set a path in Debian
+#set path = ($path /usr/local/samba/bin)
 
-set smbpasswd = /usr/local/samba/private/smbpasswd
-set user_map  = /usr/local/samba/lib/users.map
+set smbpasswd = /etc/samba/smbpasswd
+set user_map  = /etc/samba/users.map
 #
 # Set to site specific passwd command
 #
