--- samba-2.2.3a/source/rpc_server/srv_spoolss_nt.c.orig	Sat Feb  2 19:46:52 2002
+++ samba-2.2.3a/source/rpc_server/srv_spoolss_nt.c	Wed Feb 20 13:48:01 2002
@@ -4438,6 +4438,8 @@
 	return True;
 }
 
+#if 0	/* JERRY */
+
 /* Return true if two devicemodes are equal */
 
 #define DEVMODE_CHECK_INT(field) \
@@ -4447,6 +4449,10 @@
         return False; \
     }
 
+/************************************************************************
+ Handy, but currently unused functions
+ ***********************************************************************/
+
 static BOOL nt_devicemode_equal(NT_DEVICEMODE *d1, NT_DEVICEMODE *d2)
 {
 	if (!d1 && !d2) goto equal;  /* if both are NULL they are equal */
@@ -4678,6 +4684,8 @@
 	return True;
 }
 
+#endif
+
 /********************************************************************
  * Called by spoolss_api_setprinter
  * when updating a printer description.
@@ -4751,9 +4759,13 @@
 		goto done;
 	}
 
-	/* NT likes to call this function even though nothing has actually
-	   changed.  Check this so the user doesn't end up with an
-	   annoying permission denied dialog box. */
+#if 0	/* JERRY */
+	
+	/*
+	 * Another one of those historical misunderstandings...
+	 * This is reminisent of a similar call we had in _spoolss_setprinterdata()
+	 * I'm leaving it here as a reminder.  --jerry
+	 */
 
 	if (nt_printer_info_level_equal(printer, old_printer)) {
 		DEBUG(3, ("update_printer: printer info has not changed\n"));
@@ -4761,6 +4773,8 @@
 		goto done;
 	}
 
+#endif
+
 	/* Check calling user has permission to update printer description */
 
 	if (!print_access_check(NULL, snum, PRINTER_ACCESS_ADMINISTER)) {
@@ -4800,7 +4814,7 @@
 		 * lookup previously saved driver initialization info, which is then
 		 * bound to the printer, simulating what happens in the Windows arch.
 		 */
-		if (strequal(printer->info_2->drivername, old_printer->info_2->drivername))
+		if (!strequal(printer->info_2->drivername, old_printer->info_2->drivername))
 			set_driver_init(printer, 2);
 	}
 
