Index: samba/dynconfig/wscript
===================================================================
--- samba.orig/dynconfig/wscript
+++ samba/dynconfig/wscript
@@ -413,7 +413,7 @@ def build(bld):
     bld.SAMBA_SUBSYSTEM('DYNCONFIG',
                         'dynconfig.c',
                         deps='replace talloc',
-                        public_headers=os_path_relpath(os.path.join(Options.launch_dir, version_header), bld.curdir),
+                        public_headers=os_path_relpath(os.path.join(Options.launch_dir, version_header), bld.curdir) + ' dynconfig.h',
                         header_path='samba',
                         cflags=cflags)
 
