Index: samba/nsswitch/winbind_client.h
===================================================================
--- samba.orig/nsswitch/winbind_client.h
+++ samba/nsswitch/winbind_client.h
@@ -25,6 +25,6 @@
 
 int winbind_write_sock(void *buffer, int count, int recursing, int need_priv);
 int winbind_read_sock(void *buffer, int count);
-void winbind_close_sock(void);
+void winbind_close_sock(void) __attribute__((destructor));
 
 const char *nss_err_str(NSS_STATUS ret);
