Description: 64 bit fix for libsmbclient
Author: Christian Perrier <bubulle@debian.org>
Bug-Debian: http://bugs.debian.org/221618
Forwarded: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=221618#27

Index: samba/source3/include/libsmbclient.h
===================================================================
--- samba.orig/source3/include/libsmbclient.h
+++ samba/source3/include/libsmbclient.h
@@ -80,6 +80,16 @@ extern "C" {
 #include <fcntl.h>
 #include <utime.h>
 
+  /* Debian bug #221618 */
+#ifdef _LARGEFILE64_SOURCE
+#undef _LARGEFILE64_SOURCE
+#endif
+#define _LARGEFILE64_SOURCE 1
+#ifdef _FILE_OFFSET_BITS
+#undef _FILE_OFFSET_BITS
+#endif
+#define _FILE_OFFSET_BITS 64
+
 #define SMBC_BASE_FD        10000 /* smallest file descriptor returned */
 
 #define SMBC_WORKGROUP      1
