Index: samba/source4/rpc_server/dcerpc_server.c
===================================================================
--- samba.orig/source4/rpc_server/dcerpc_server.c	2015-04-24 13:29:56.803570042 +0200
+++ samba/source4/rpc_server/dcerpc_server.c	2015-04-24 13:31:03.199573895 +0200
@@ -500,6 +500,7 @@
 
 	if (c->iface && c->iface->unbind) {
 		c->iface->unbind(c, c->iface);
+		c->iface = NULL;
 	}
 
 	return 0;
@@ -616,6 +617,16 @@
 		extra_flags |= DCERPC_PFC_FLAG_SUPPORT_HEADER_SIGN;
 	}
 
+	if ((call->pkt.pfc_flags & DCERPC_PFC_FLAG_CONC_MPX) &&
+            (call->state_flags & DCESRV_CALL_STATE_FLAG_MULTIPLEXED)) {
+                call->context->conn->state_flags |= DCESRV_CALL_STATE_FLAG_MULTIPLEXED;
+                extra_flags |= DCERPC_PFC_FLAG_CONC_MPX;
+        }
+
+	if (call->state_flags & DCESRV_CALL_STATE_FLAG_PROCESS_PENDING_CALL) {
+                call->context->conn->state_flags |= DCESRV_CALL_STATE_FLAG_PROCESS_PENDING_CALL;
+        }
+
 	/* handle any authentication that is being requested */
 	if (!dcesrv_auth_bind(call)) {
 		talloc_free(call->context);
@@ -795,6 +806,7 @@
 	NTSTATUS status;
 	uint32_t result=0, reason=0;
 	uint32_t context_id;
+	uint32_t extra_flags = 0;
 
 	/* handle any authentication that is being requested */
 	if (!dcesrv_auth_alter(call)) {
@@ -828,12 +840,22 @@
 		reason = DCERPC_BIND_REASON_ASYNTAX;
 	}
 
+	if ((result == 0 && call->pkt.pfc_flags & DCERPC_PFC_FLAG_CONC_MPX)) {
+                if (call->context->conn->state_flags & DCESRV_CALL_STATE_FLAG_MULTIPLEXED) {
+                        extra_flags |= DCERPC_PFC_FLAG_CONC_MPX;
+                }
+        }
+
+        if (result == 0 && call->state_flags & DCESRV_CALL_STATE_FLAG_PROCESS_PENDING_CALL) {
+                call->context->conn->state_flags |= DCESRV_CALL_STATE_FLAG_PROCESS_PENDING_CALL;
+        }
+
 	/* setup a alter_resp */
 	dcesrv_init_hdr(&pkt, lpcfg_rpc_big_endian(call->conn->dce_ctx->lp_ctx));
 	pkt.auth_length = 0;
 	pkt.call_id = call->pkt.call_id;
 	pkt.ptype = DCERPC_PKT_ALTER_RESP;
-	pkt.pfc_flags = DCERPC_PFC_FLAG_FIRST | DCERPC_PFC_FLAG_LAST;
+	pkt.pfc_flags = DCERPC_PFC_FLAG_FIRST | DCERPC_PFC_FLAG_LAST | extra_flags;
 	pkt.u.alter_resp.max_xmit_frag = 0x2000;
 	pkt.u.alter_resp.max_recv_frag = 0x2000;
 	if (result == 0) {
@@ -1307,6 +1329,18 @@
 		cur = next;
 		next = cur->next;
 
+		if (cur->state_flags & DCESRV_CALL_STATE_FLAG_PROCESS_PENDING_CALL) {
+                        struct dcesrv_connection_context *context_cur, *context_next;
+ 
+                        context_next = cur->contexts;
+                        while (context_next != NULL) {
+                                context_cur = context_next;
+                                context_next = context_cur->next;
+ 
+                                dcesrv_connection_context_destructor(context_cur);
+                        }
+                }
+
 		dcesrv_terminate_connection(cur, cur->terminate);
 	}
 }
Index: samba/source4/rpc_server/dcerpc_server.h
===================================================================
--- samba.orig/source4/rpc_server/dcerpc_server.h	2015-04-24 13:29:56.803570042 +0200
+++ samba/source4/rpc_server/dcerpc_server.h	2015-04-24 13:29:56.799570042 +0200
@@ -102,6 +102,8 @@
 #define DCESRV_CALL_STATE_FLAG_ASYNC (1<<0)
 #define DCESRV_CALL_STATE_FLAG_MAY_ASYNC (1<<1)
 #define DCESRV_CALL_STATE_FLAG_HEADER_SIGNING (1<<2)
+#define DCESRV_CALL_STATE_FLAG_MULTIPLEXED (1<<3)
+#define DCESRV_CALL_STATE_FLAG_PROCESS_PENDING_CALL (1<<4)
 	uint32_t state_flags;
 
 	/* the time the request arrived in the server */
