diff --git a/ctdb/wscript b/ctdb/wscript
index 06f86a3..055c853 100755
--- a/ctdb/wscript
+++ b/ctdb/wscript
@@ -94,7 +94,8 @@ def configure(conf):
     conf.RECURSE('lib/talloc')
     conf.RECURSE('lib/tevent')
     conf.RECURSE('lib/tdb')
-    conf.RECURSE('lib/socket_wrapper')
+    if conf.CONFIG_GET('ENABLE_SELFTEST'):
+        conf.RECURSE('lib/socket_wrapper')
 
     conf.CHECK_HEADERS('sched.h')
     conf.CHECK_HEADERS('procinfo.h')
@@ -256,7 +257,8 @@ def build(bld):
     bld.RECURSE('lib/talloc')
     bld.RECURSE('lib/tevent')
     bld.RECURSE('lib/tdb')
-    bld.RECURSE('lib/socket_wrapper')
+    if bld.CONFIG_GET('ENABLE_SELFTEST'):
+        bld.RECURSE('lib/socket_wrapper')
 
     if bld.env.standalone_ctdb:
         # In a combined build is implemented, CTDB will wanted to
