Index: samba-3.0.23c/source/script/installswat.sh
===================================================================
--- samba-3.0.23c.orig/source/script/installswat.sh	2006-09-02 10:08:47.448570790 +0200
+++ samba-3.0.23c/source/script/installswat.sh	2006-09-02 10:09:19.024825937 +0200
@@ -194,7 +194,11 @@
 
 # Install/ remove Using Samba book (but only if it is there)
 
-if [ "x$BOOKDIR" != "x" -a -f $SRCDIR../docs/htmldocs/using_samba/toc.html ]; then
+# Under Debian we don't actually install the book. The book is part of
+# the samba-doc package, so we just provide a symlink that points to
+# where the book is actually installed. The symlink is created in
+# debian/rules.
+if /bin/false; then
 
     # Create directories
 
