Index: samba/source4/auth/ntlm/auth.c
===================================================================
--- samba.orig/source4/auth/ntlm/auth.c
+++ samba/source4/auth/ntlm/auth.c
@@ -279,7 +279,7 @@ _PUBLIC_ struct tevent_req *auth_check_p
 	state->user_info	= user_info;
 
 	if (!user_info->mapped_state) {
-		nt_status = map_user_info(auth_ctx->sam_ctx, req, lpcfg_workgroup(auth_ctx->lp_ctx),
+		nt_status = map_user_info(auth_ctx->sam_ctx, req, auth_ctx->lp_ctx, lpcfg_workgroup(auth_ctx->lp_ctx),
 					  user_info, &user_info_tmp);
 		if (tevent_req_nterror(req, nt_status)) {
 			return tevent_req_post(req, ev);
Index: samba/source4/auth/ntlm/auth_util.c
===================================================================
--- samba.orig/source4/auth/ntlm/auth_util.c
+++ samba/source4/auth/ntlm/auth_util.c
@@ -43,6 +43,7 @@ NTSTATUS auth_get_challenge_not_implemen
 ****************************************************************************/
 static NTSTATUS map_user_info_cracknames(struct ldb_context *sam_ctx,
 					 TALLOC_CTX *mem_ctx,
+					 struct loadparm_context *lp_ctx,
 					 const char *default_domain,
 					 const struct auth_usersupplied_info *user_info,
 					 struct auth_usersupplied_info **user_info_mapped)
@@ -64,7 +65,8 @@ static NTSTATUS map_user_info_cracknames
 
 	/* use cracknames to work out what domain is being
 	   asked for */
-	if (strchr_m(user_info->client.account_name, '@') != NULL) {
+	if (strchr_m(user_info->client.account_name, '@') != NULL
+	    && !lpcfg_parm_bool(lp_ctx, NULL, "auth", "usernames are emails", false)) {
 		werr = DsCrackNameOneName(sam_ctx, tmp_ctx, 0,
 					  DRSUAPI_DS_NAME_FORMAT_USER_PRINCIPAL,
 					  DRSUAPI_DS_NAME_FORMAT_NT4_ACCOUNT,
@@ -222,6 +224,7 @@ static NTSTATUS map_user_info_cracknames
 ****************************************************************************/
 NTSTATUS map_user_info(struct ldb_context *sam_ctx,
 		       TALLOC_CTX *mem_ctx,
+		       struct loadparm_context *lp_ctx,
 		       const char *default_domain,
 		       const struct auth_usersupplied_info *user_info,
 		       struct auth_usersupplied_info **user_info_mapped)
@@ -234,7 +237,7 @@ NTSTATUS map_user_info(struct ldb_contex
 	if (sam_ctx != NULL) {
 		/* if possible, use cracknames to parse the
 		   domain/account */
-		return map_user_info_cracknames(sam_ctx, mem_ctx, default_domain, user_info, user_info_mapped);
+		return map_user_info_cracknames(sam_ctx, mem_ctx, lp_ctx, default_domain, user_info, user_info_mapped);
 	}
 
 	DEBUG(0,("map_user_info: Mapping user [%s]\\[%s] from workstation [%s] default_domain=%s\n",
