diff -uNr samba-3.0.14a.orig/source/rpc_server/srv_pipe.c samba-3.0.14a/source/rpc_server/srv_pipe.c
--- samba-3.0.14a.orig/source/rpc_server/srv_pipe.c	2005-04-13 23:14:24.000000000 -0700
+++ samba-3.0.14a/source/rpc_server/srv_pipe.c	2005-07-12 00:51:04.000000000 -0700
@@ -158,7 +158,7 @@
 
 	if(p->out_data.data_sent_length + data_len >= prs_offset(&p->out_data.rdata)) {
 		p->hdr.flags |= RPC_FLG_LAST;
-		if ((auth_seal || auth_verify) && (data_len_left % 8)) {
+		if ((auth_seal || auth_verify || p->netsec_auth_validated) && (data_len_left % 8)) {
 			ss_padding_len = 8 - (data_len_left % 8);
 			DEBUG(10,("create_next_pdu: adding sign/seal padding of %u\n",
 				ss_padding_len ));
