Index: samba-3.0.23c/source/client/smbmount.c
===================================================================
--- samba-3.0.23c.orig/source/client/smbmount.c	2006-09-02 10:09:32.432934252 +0200
+++ samba-3.0.23c/source/client/smbmount.c	2006-09-02 10:09:33.220940617 +0200
@@ -211,6 +211,10 @@
   		c->capabilities &= ~CAP_STATUS32;
 		c->force_dos_errors = True;
 	}
+	/* For now, respect requests for uid-flattening mount options
+	   by disabling Unix permissions handling in the kernel driver */
+	if (mount_uid || mount_gid || mount_fmask || mount_dmask)
+		c->capabilities &= ~CAP_UNIX;
 
 	if (!cli_session_setup(c, username, 
 			       password, strlen(password),
