Index: samba-3.0.25rc2/source/client/smbmount.c
===================================================================
--- samba-3.0.25rc2.orig/source/client/smbmount.c	2007-04-24 11:07:54.208043743 +0200
+++ samba-3.0.25rc2/source/client/smbmount.c	2007-04-24 11:07:54.772062980 +0200
@@ -213,6 +213,10 @@
   		c->capabilities &= ~CAP_STATUS32;
 		c->force_dos_errors = True;
 	}
+	/* For now, respect requests for uid-flattening mount options
+	   by disabling Unix permissions handling in the kernel driver */
+	if (mount_uid || mount_gid || mount_fmask || mount_dmask)
+		c->capabilities &= ~CAP_UNIX;
 
 	if (!NT_STATUS_IS_OK(cli_session_setup(c, username, 
 					       password, strlen(password),
