Description: Avoid using bashism in smbtar
Author: Jelmer Vernooij <jelmer@samba.org>
Bug-Debian: http://bugs.debian.org/486056
Forwarded: no

Index: b/source3/script/smbtar
===================================================================
--- a/source3/script/smbtar	2009-11-08 15:54:27.000000000 +0100
+++ b/source3/script/smbtar	2009-11-08 15:55:36.000000000 +0100
@@ -151,7 +151,7 @@
 if [ -z "$verbose" ]; then
       echo "server    is $server"
 #     echo "share     is $service"
-      echo "share     is $service\\$cdcmd"
+      printf "share     is %s\\%s\n" "$service" "$cdcmd"
       echo "tar args  is $tarargs"
 #     echo "password  is $password"  # passwords should never be sent to screen
       echo "tape      is $tapefile"
